modules/siddhi-core/src/main/java/org/wso2/siddhi/core/aggregation/IncrementalDataPurging.java

```diff
@@ -0,0 +1,366 @@
+        } else {
+            throw new NoSuchDurationException("Provided retention parameter cannot be identified. retention period:  "
+                    + value + ".");
+        }
+    }
+
+    /**
+     * normalize user provided time durations
+     */
+    private TimePeriod.Duration durationNormalization(String value) {
+        switch (value) {
+            case "sec":
+            case "seconds":
+            case "second":
+                return TimePeriod.Duration.SECONDS;
+            case "min":
+            case "minutes":
+            case "minute":
+                return TimePeriod.Duration.MINUTES;
+            case "h":
+            case "hour":
+            case "hours":
+                return TimePeriod.Duration.HOURS;
+            case "days":
+            case "day":
+                return TimePeriod.Duration.DAYS;
+            case "month":
+            case "months":
+                return TimePeriod.Duration.MONTHS;
+            case "year":
+            case "years":
+                return TimePeriod.Duration.YEARS;
+            default:
+                throw new NoSuchDurationException("Duration '" + value + "' does not exists ");
+        }
+    }
+
+    /**
+     * Building the MatchingMetaInfoHolder for delete records
+     **/
+    private MatchingMetaInfoHolder matchingMetaInfoHolder(Table table, Attribute attribute) {
+        MetaStateEvent metaStateEvent = new MetaStateEvent(2);
+
+        MetaStreamEvent metaStreamEventWithDeletePara = new MetaStreamEvent();
+        MetaStreamEvent metaStreamEventForTable = new MetaStreamEvent();
+
+        TableDefinition deleteTableDefinition = TableDefinition.id("");
+        deleteTableDefinition.attribute(attribute.getName(), attribute.getType());
+        metaStreamEventWithDeletePara.setEventType(MetaStreamEvent.EventType.TABLE);
+        metaStreamEventWithDeletePara.addOutputData(attribute);
+        metaStreamEventWithDeletePara.addInputDefinition(deleteTableDefinition);
+
+
+        metaStreamEventForTable.setEventType(MetaStreamEvent.EventType.TABLE);
+        for (Attribute attributes : table.getTableDefinition().getAttributeList()) {
+            metaStreamEventForTable.addOutputData(attributes);
+        }
+        metaStreamEventForTable.addInputDefinition(table.getTableDefinition());
+
+        metaStateEvent.addEvent(metaStreamEventWithDeletePara);
+        metaStateEvent.addEvent(metaStreamEventForTable);
+
+        TableDefinition definition = table.getTableDefinition();
+        return new MatchingMetaInfoHolder(metaStateEvent,
+                0, 1, deleteTableDefinition, definition, 0);
+    }
+
+    /**
+     * Data purging task scheduler method
+     **/
+    public static void executeIncrementalDataPurging(SiddhiAppContext siddhiAppContext,
+                                                     IncrementalDataPurging incrementalDataPurging) {
+
+        if (Objects.nonNull(scheduledFuture)) {
+            scheduledFuture.cancel(true);
```

Comment:
Why is this ?

if we purge sec and min table at close proximity dont they purge parallelly ?

Reply:
purging for all the tables will schedule as one single task, so by doing this, above mentioned scenario will not get affected. the intention of doing this is because when we deploy an aggregation siddhi app with purging on the editor and if we change the aggregator name, it will keep the old scheduled jobs and schedule a new job for new tables. To avoid that I'm canceling previously scheduled job in each time we reinitialize the aggregation runtime and reinitialize new job.